ocean_lib/data_provider/data_service_provider.py

```diff
@@ -103,7 +102,108 @@ def encrypt(
 
     @staticmethod
     @enforce_types
-    def sign_message(wallet: Wallet, msg: str) -> str:
+    def initialize(
+        did: str,
+        service_id: str,
+        consumer_address: str,
+        service_endpoint: str,
+        userdata: Optional[Dict] = None,
+    ) -> Response:
+
+        req = PreparedRequest()
+        service_endpoint += f"?documentId={did}"
+        payload = {
+            "serviceId": service_id,
+            "consumerAddress": consumer_address,
+        }
+
+        if userdata:
+            userdata = json.dumps(userdata)
+            payload["userdata"] = userdata
+
+        req.prepare_url(service_endpoint, payload)
+
+        response = DataServiceProvider._http_method("get", req.url)
+
+        if not response or not hasattr(response, "status_code"):
+            raise Exception("Response not found!")
+
+        if response.status_code != 200:
+            msg = (
+                f"Initialize service failed at the initializeEndpoint "
+                f"{req.url}, reason {response.text}, status {response.status_code}"
+            )
+            logger.error(msg)
+            raise Exception(msg)
+
+        logger.info(
+            f"Service initialized successfully" f" initializeEndpoint {req.url}"
+        )
+
+        return response
+
+    @staticmethod
+    @enforce_types
+    def download(
+        did: str,
+        service_id: str,
+        tx_id: str,
+        files: List[Dict[str, Any]],
```

Comment:
please revert this change of passing the files dict. We are just blindly downloading files from an asset. I am not the publisher, files are encrypted, I don't want to call fileinfo. I bought the asset and I want to download it. I want to have the option of not passing in files dicts, files indices, file lengths etc.

-> Yes, I _could_ have knowledge and _choose_ to pass in the file index, as this was previously implemented. But not file objects and stuff like that. This is a library to help people use OCEAN, not make them use our File Object etc. Those are for our testing purposes and for "advanced" users. In my previous review, I asked to have fileIndex as an int or as None. Not to make this a file object, dict or something else.

Reply:
I have the following problem: provider download endpoint requires `fileIndex` field, otherwise it will throw an error (https://github.com/oceanprotocol/provider/blob/v4main/ocean_provider/routes/consume.py#L282). As you mentioned above, files are encrypted and we don't want to pass a list of files. In this case, I suggest to implement the `fileinfo` endpoint and call it inside the download function from `DataServiceProvider` class, so that the user is not obliged to call it separately. I think this is the easiest way to implement the proposed functionality avoiding passing the file list.
BTW, in V3, the file list was passed to the download service as it is shown [here](https://github.com/oceanprotocol/ocean.py/blob/main/ocean_lib/data_provider/data_service_provider.py#L197) and there were only 2 fields from the payload that differs from the V4 one. @DMats, @alexcos20, what do you think?

Reply:
UPDATE: The solution will be to implement `fileinfo` endpoint in ocean.py and use it inside the download endpoint when the parameter `index` is None.